dvc/utils/__init__.py

```diff
@@ -262,7 +262,12 @@ def fix_env(env=None):
             # NOTE: py2 doesn't like unicode strings in environ
             env[cast_bytes_py2(lp_key)] = cast_bytes_py2(lp_orig)
         else:
-            env.pop(lp_key, None)
+            # We used to delete "LD_LIBRARY_PATH" key. We cannot do that
+            # because GitPython uses env to "update" his own internal state.
+            # When there is no key in env, this value is not updated and
```

Comment:
I guess this part of the explanation is about why setting `None` doesn't work, no?

Reply:
I wouldn't say that. 
This explains why we prefer to set the empty string for "LD_LIBRARY_PATH" over deleting it from the `env` dictionary.

Reply:
Also please add a link to that gitpython issue.

Reply:
And it is worth noting that this is only relevant for `clone_from` really, where `env` behaves weirdly, but other things work fine. Hm, it also makes me think whether we should do needed modifications for clone_from specifically and not here. I don't see how this approach would backstub us in the future yet, but it sure makes me a bit nervous :smiley:

Reply:
Discussed with @pared that it would make sense to indeed move this logic to dvc/scm/git/__init__.py:Git.clone to be safe. It could look something like
```
env = fix_env()
if is_binary():
    env["LD_LIBRARY_PATH"] = ""
```